### DISCUSSION TITLE
Any plans to update to latest SDK ?

### DISCUSSION THREAD

**Discussion by takhimi** [Poing-Studios/Godot-AdMob-Android-iOS#117] [General]
Hi. Currently the Admob  sdk and the UMP sdk is outdated. Do you plan to update to latest sdk for them ? Any ETA ?

Thanks

**Comment by gumaciel** [comment]
Thank you for asking this, i plan to update on this friday

**Reply by takhimi** [reply]
Awesome. Thank you.

**Reply by gumaciel** [reply]
@takhimi 

Released on Version v2.0.1

https://github.com/Poing-Studios/godot-admob-android/releases/tag/v2.0.1

**Comment by takhimi** [comment]
Thank you for the updates and for all your works.

**Comment by takhimi** [comment]
Is it possible to update the UMP sdk to version 2.0 also ? I think it is still at version 1.0 for the UMP.

**Reply by gumaciel** [reply]
Yes, sorry for that i didn't saw the version 2.0 for UMP 

I will generate again the version

**Reply by takhimi** [reply]
No problem and thank you for your work.

**Comment by m-melis** [comment]
After the latest version update `compileSdk` 31 seems required, which is not available from Godot 3.4.4. After manually editing `config.gradle` works fine anyway.

Thanks for your work btw.

**Reply by takhimi** [reply]
> After the latest version update `compileSdk` 31 seems required, which is not available from Godot 3.4.4. After manually editing `config.gradle` works fine anyway.
> 
> Thanks for your work btw.

I think we actually should start to target for 31 as soon as possible since it is required for android 12 starting this august. https://developer.android.com/google/play/requirements/target-sdk

**Reply by gumaciel** [reply]
Hi guys, just checked now and i think it's not required to update the `compileSdk` to 31, the only think i'm seeing is needed to update it's the `minSdkVersion` to 19

https://developers.google.com/admob/android/migration#minsdkversion

**Reply by gumaciel** [reply]
Just checked now your link @takhimi , i will require this change to Godot team to enter on new Godot versions, also i will update our `config.gradle`

**Reply by gumaciel** [reply]
Done: https://github.com/godotengine/godot/pull/62297

**Reply by gumaciel** [reply]
Hello guys @takhimi , @m-melis , i generate a new version of the Plugin with the fix of UMP and Gradle, please test it if nothing breaks: 

https://github.com/Poing-Studios/godot-admob-android/releases/tag/v2.0.1

**Comment by takhimi** [comment]
@gumaciel , thank you for the update. However right now I'm a bit demotivated with admob and ump. It turns out user or gamer can easily disable the ads or make the ads not shown anymore by simply click on "Manage options" then after that just click "Submit"(I think all options will be turn off by default). Google wont even serve any ads to the user. A lot of people have complaint about this, 

https://groups.google.com/g/google-admob-ads-sdk/c/UcveWmtBm4Q/m/Fb-rtp2vAwAJ
-you probably need to click on the response name to see their messages.

I think google is slowly pushing us to just use IAP in future...

The good news is that if we use UMP2.0 it maybe will serve Limited ads (not the non personalized ads). My guess that this is probably some very low ecpm or some cheap generic ads or something.  https://support.google.com/admob/answer/10105530

So next august we need to target sdk 31 for Android 12 phones, declare the data collection policy and don't forget about the new Ad-serving protections for teens here
https://support.google.com/admob/answer/12171027

Which they will be enforcing this august also. Basically any gamer that is age below 18 years but above 13 years, we maybe only allowed to serve 'G' ads and it will be non personalized only. But I think it is set automatically and it will depends on the Google account at the user phone. We are going to have low ecpm for sure starting this august with teens gamer.

Right now at my google dashboard I just select my target audience to be 18+ years only for all my games. Because I'm tired of all these rules and I dont see it worth my time to make games for all group with ads anymore. Probably should start focusing on creating more complex games that targeting older player and using IAP more ingame. 

Maybe for admob we need to implement the SSV callback for rewarded ads for free ingame coins or something because it is easily hackable without ssv callback.

**Reply by gumaciel** [reply]
Sorry for that, I believe that's not our fault.

If ads aren't showing for you when it's "non-personalized", don't worry because it should show up for the user:

Please read this Issue: https://github.com/Poing-Studios/godot-admob-android/discussions/102

```
if the error is "ERROR_CODE_NO_FILL", then that mean implantation is correct but there is no ads available in your area at the moment

try using a VPN or creating an ad yourself on admob.com, remember that it may take some time for ads to work if they are created on admob.com

```
Read more here: https://support.google.com/admob/answer/9469204?hl=en
And here: https://groups.google.com/g/google-admob-ads-sdk/c/xfLEdy06pyc

I agree with you that these data rules are pretty boring and that they make us lose money on advertising, I think the biggest blame for this is on world governments, especially Europe that started with GDPR and is now going to the whole world, that already is present in Brazil with the LGPD.

Unfortunately I think that all mobile advertising (AdMob, Facebook, UnityAds, AdColony etc..) will adopt these measures that AdMob is implementing.


`Maybe for admob we need to implement the SSV callback for rewarded ads for free ingame coins or something because it is easily hackable without ssv callback.`

I hadn't heard of this before, if possible open another discussion about it or join our Discord: https://discord.gg/YEPvYjSSMk

**Comment by takhimi** [comment]
@gumaciel you don't need to apologize or anything. It is just my rant to my problems and to Google. Also I think you misunderstood me, I'm not saying that the ads wont appear because there is no fill from google. I'm saying that Google wont show any ads if at least 3 important categories are selected in the "manage options" which is disabled by default when user open it. 

https://stackoverflow.com/questions/63464270/how-to-request-non-personalized-admob-ads-with-the-new-unified-messaging-platfor/64654352#64654352

**Reply by gumaciel** [reply]
I think I understand what you mean, but I think the error that gives is the `ERROR_CODE_NO_FILL` (If this is not the error, what is the error you see when you try to load an ad?) when you refuse or just accept some consent options, see this user's response from the forum link I sent you:

https://groups.google.com/g/google-admob-ads-sdk/c/xfLEdy06pyc/m/yy2DDc5rAAAJ

![Captura de Tela 2022-06-23 às 21 17 51](https://user-images.githubusercontent.com/20030153/175435862-07e35bb9-cfed-483a-b974-8578f7cca492.png)

![Captura de Tela 2022-06-23 às 21 23 36](https://user-images.githubusercontent.com/20030153/175436300-c1a6955c-fef8-4130-92bd-6dab97166fb9.png)

**Reply by gumaciel** [reply]
Btw soon i will test again the UMP with this new version (v2.0.0) to see if it also happens to me, but i need to configure somethings to work for me.

**Reply by gumaciel** [reply]
> which is disabled by default when user open it.

there is no way to mark them to be already enable for the user in `Privacy & messaging`?

**Reply by takhimi** [reply]
I don't think we can change the default consent options. I already tried it before this. We can basically just change the theme of the consent form.

**Comment by gumaciel** [comment]
@takhimi Theres a big issue here:

After I was testing the new version 2.0.1 of Android Plugin, I found that it is giving this error:

![Captura de Tela 2022-06-24 às 00 51 55](https://user-images.githubusercontent.com/20030153/175458819-25b44aec-0f33-468b-8ec2-2c2ce672978d.png)

Checking the Release Notes, I found this from this release:
![Captura de Tela 2022-06-24 às 01 31 00](https://user-images.githubusercontent.com/20030153/175462549-4ddd9576-4b47-4fca-b58d-8eb150c4dff6.png)

https://developers.google.com/admob/android/rel-notes

---
So for that, I will need to cancel this upgrade to version 21.0.0 of the Mobile Ads SDK, the most I can update will be to version 20.4.0

For versions 20.5.0 onwards, it will be necessary to use only the Godot versions that will arrive (3.5; 3.4.5; 3.3.5; 4.0; etc...)

That said, I believe this version upgrade to 21.0.0 might take a little longer as I would like to add [mediation](https://developers.google.com/admob/android/mediate) to as many versions as possible for Godot.

Before doing anything, could you check for me if this error also happens to you? The error only disappears when I use version 20.4.0 and below. Thank you!

---
Edit: As I saw, we will can use UMP 2.0.0, but the Mobile Ads SDK will need to be 20.4.0 for now

**Reply by takhimi** [reply]
@gumaciel 

It works just fine with my phone ?  I'm using the Godot 3.5 rc4 mono and c#. 
The consent and the ads dont have any problem or errors.

I even deleted and redownload back the android build template just to be sure.

However maybe because I also included the Google Play Services for the achievement and etc ?

Oddly I dont need to change the targetsdk ?

`ext.versions = [
    androidGradlePlugin: '7.0.3',
    compileSdk         : 31,
    minSdk             : 19, // Also update 'platform/android/java/lib/AndroidManifest.xml#minSdkVersion' & 'platform/android/export/export_plugin.cpp#DEFAULT_MIN_SDK_VERSION'
    targetSdk          : 30, // Also update 'platform/android/java/lib/AndroidManifest.xml#targetSdkVersion' & 'platform/android/export/export_plugin.cpp#DEFAULT_TARGET_SDK_VERSION'
    buildTools         : '30.0.3',
    kotlinVersion      : '1.6.21',
    fragmentVersion    : '1.3.6',
    nexusPublishVersion: '1.1.0',
    javaVersion        : 11,
    ndkVersion         : '21.4.7075529' // Also update 'platform/android/detect.py#get_project_ndk_version()' when this is updated.`



The admob.gdap shows the latest one.


remote=["com.google.android.gms:play-services-ads:21.0.0", "com.google.android.ump:user-messaging-platform:2.0.0"]

**Reply by gumaciel** [reply]
@takhimi I just tested on 3.5 RC4 and it's working.

I tested with the [3.4.4-stable build](https://github.com/Poing-Studios/godot-admob-android/releases/download/v2.0.1/android-standard-template-v3.4.4.zip), but i don't recommend you do that because it can cause some bugs.

To generate the build we get this `.aar` file: 
https://downloads.tuxfamily.org/godotengine/3.4.4/godot-lib.3.4.4.stable.release.aar
https://downloads.tuxfamily.org/godotengine/3.3/godot-lib.3.3.stable.release.aar

As you can see, it has some difference between the size of files, the plugin can work for you if you use from different version, but might cause unexpected bug.

---

> Oddly I dont need to change the targetsdk ?

I don't know, this is from Godot source code? I guess not.

---

So i will generate again the version with the Mobile Ads SDK on 20.4.0 because it is broken on older versions.

**Reply by takhimi** [reply]
Ok. Thank you. Good to know. But if you want to support for Android 12 this August, you need to use at least SDK 20.5.0 and to support for the Limited Ads you need at least SDK 20.6.0.

https://developers.google.com/admob/android/rel-notes

**Reply by gumaciel** [reply]
thanks for reminding me, as soon as a version of Godot comes out that supports this update to 21.0.0 I will do that
